system/smp.c

```diff
@@ -682,11 +725,26 @@ static bool parse_slit(uintptr_t slit_addr)
 }
 #endif
 
-static inline void send_ipi(int apic_id, int trigger, int level, int mode, uint8_t vector)
+static inline void send_ipi(int apic_id, int trigger __attribute__((unused)), int level __attribute__((unused)), int mode, uint8_t vector)
 {
+#if defined(__i386__) || defined(__x86_64__)
     apic_write(APIC_REG_ICRHI, apic_id << 24);
 
     apic_write(APIC_REG_ICRLO, trigger << 15 | level << 14 | mode << 8 | vector);
+#elif defined(__loongarch_lp64)
+    if (mode == APIC_DELMODE_STARTUP) {
+        //
+        // Set the AP mailbox0
+        //
+        __iocsrwr_d((1ULL << 31 | ((0x0 << 1) + 1) << 2 | apic_id << 16 | (((uintptr_t)ap_startup_addr) & 0xFFFFFFFF00000000ULL)), 0x1048);
+        __iocsrwr_d((1ULL << 31 | (0x0 << 1) << 2 | apic_id << 16 | (((uintptr_t)ap_startup_addr << 32))), 0x1048);
+    }
+
+    //
+    // Trigger IPI interrupt
```

Comment:
"IPI" is "inter-processor interrupt". ;-)
```suggestion
    // Trigger IPI
```